File: javaCode/SmithGame/src/me/haved/engine/scene/Scene.java
package me.haved.engine.scene;

import me.haved.engine.math.Matrix4;

public class Scene {
	private GameObject rootObject;
	
	public Scene() {
		rootObject = new GameObject();
		rootObject.SetScene(this);
	}
	
	public void Update() {
		rootObject.Update();
	}
	
	public void Render(Matrix4 VP) {
		rootObject.Render(VP);
	}
	
	public void AddGameObject(GameObject object) {
		rootObject.AddChild(object);
	}
}

File: cppCode/SmithGame/src/scene/Scene.cpp
#include "Scene.h"

void Scene::AddObject(GameObject& object) {
    root.AddChild(object);
    object.SetScene(this);
}

void Scene::Update() {
    root.Update();
}

void Scene::Render(glm::mat4 &VP) {
    root.Render(VP);
}
